hyperspy/model.py

```diff
@@ -49,6 +49,17 @@
 from hyperspy.gui.tools import ComponentFit
 from hyperspy.component import Component
 from hyperspy.signal import Signal
+from hyperspy.misc.utils import slugify
+
+
+class ModelComponents(object):
+
+    """Container for model components.
+
+    Useful to provide tab completion when running in IPython.
+
+    """
+    pass
```

Comment:
How about adding `__repr__` method, where each component is printed in a new line?
I personally find the current printing confusing, as all the extra characters (while useful to determine actual class of the component) clutter the print statement and it takes extra time to figure out where components start and end